### PR TITLE
LDSv2 bug fix

### PR DESCRIPTION
**Pull request by rshriram**
combine multiple vhosts into the same route configuration.

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -290,45 +291,47 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
...
+				}
+
 				listenerOpts.networkFilters = buildOutboundNetworkFilters(clusterName, addresses, servicePort)
-				l = buildListener(listenerOpts)
-				tcpListeners = append(tcpListeners, l)
-				// TODO: Set SNI for HTTPS
 			case model.ProtocolHTTP2, model.ProtocolHTTP, model.ProtocolGRPC:
+				listenerMapKey = fmt.Sprintf("%s:%d", listenAddress, servicePort.Port)
+				if l, exists := listenerMap[listenerMapKey]; exists {
+					if !strings.HasPrefix(l.Name, "http") {
```
Comment (ijsnellf) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: this is brittle. maybe we should store the protocol etc along with the listener.

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 2]
```diff
@@ -190,7 +190,8 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env model.Environmen
 		// TODO: need inbound listeners in HTTP_PROXY case, with dedicated ingress listener.
 	}
 
-	return util.NormalizeListeners(listeners), nil
+	return listeners, nil
+	//return util.NormalizeListeners(listeners), nil
```
Comment (ijsnellf) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: we don't want to normalize listeners?

Comment (rshriram) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: No. That code is not right.

File: pilot/pkg/networking/core/v1alpha3/listener.go (hunk 1 above)
Comment (rshriram) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: Yeah I didn’t like it either. Will try to rework in next iteration.

**Comment by rshriram** [comment]
merging due to CLA issue

**Pull request merged**